src/hooks/useAffiliatesInfo.ts

```diff
@@ -38,11 +50,31 @@ export const useAffiliatesInfo = (dydxAddress?: string) => {
     }
   };
 
-  const query = useQuery({
+  const affiliateMetadataQuery = useQuery({
     queryKey: ['affiliatesMetadata', dydxAddress],
-    queryFn,
+    queryFn: fetchAffiliateMetadata,
     enabled: Boolean(compositeClient && dydxAddress),
   });
 
-  return query;
+  const fetchAffiliateMaxEarning = async () => {
+    const allAffiliateTiers = await getAllAffiliateTiers();
+    let maxRevshare = DEFAULT_MAX_AFFILIATE_SHARE;
+    const lastTier = allAffiliateTiers?.at(-1);
+    if (lastTier) {
+      maxRevshare = lastTier.takerFeeSharePpm / 1_000_000;
+    }
```

Comment:
you can probably simplify this now:
```suggestion
    const lastTier = allAffiliateTiers?.at(-1);
    const maxRevShare = lastTier ? lastTier.takerFeeSharePpm / 1_000_000 : DEFAULT_MAX_AFFILIATE_SHARE;
```

Reply:
good call